arrays/arrays.c

```diff
@@ -146,11 +146,23 @@ void arr_remove(Array *arr, char *element) {
 
   // Search for the first occurence of the element and remove it.
   // Don't forget to free its memory!
-
+  int index = 0;
+  for(int i = 0; i < arr->count; i++)
+  {
+    if(arr->elements[i] == element)
+    {
+      // arr->elements[i] == NULL;
+      free(arr->elements[i]);
+      index = i;
+    }
+  }
   // Shift over every element after the removed element to the left one position
-
+  for (int i = index; i > arr->count; i++)
```

Comment:
Examine this loop in python tutor. What happens when i = arr->count (i.e. the last loop that runs before i > arr-->count)?